ci/taos/plugins-good/pr-format-prohibited-words.sh

```diff
@@ -0,0 +1,115 @@
+# @author   Geunsik Lim <geunsik.lim@samsung.com>
+#
+
+# @brief [MODULE] TAOS/pr-format-prohibited-words
+function pr-format-prohibited-words(){
+echo -e "########################################################################################"
+echo -e "[MODULE] TAOS/pr-format-prohibited-words: Check if there are prohibited words in the text files." 
+pwd
+
+# Check if a server administrator install required commands.
+check_dependency git
+check_dependency grep
+
+# Read file names that a contributor modified(e.g., added, moved, deleted, and updated) from a last commit.
+FILELIST=`git show --pretty="format:" --name-only --diff-filter=AMRC`
+
+# Inspect all files that contributor modifed.
+for i in ${FILELIST}; do
+    # Default value of check_result is "skip".
+    check_result="skip"
+
+    # Skip obsolete folder
+    if [[ $i =~ ^obsolete/.* ]]; then
+        continue
+    fi
+    # Skip external folder
+    if [[ $i =~ ^external/.* ]]; then
+        continue
+    fi
+    # Handle only text files in case that there are lots of files in one commit.
+    echo -e "[DEBUG] file name is ( $i )."
+    if [[ `file $i | grep "ASCII text" | wc -l` -gt 0 ]]; then
+        # Run a prohibited-words module in case that a PR includes text files.
+        case $i in
+            *.c | *.h | *.cpp | *.hpp| *.py | *.sh | *.php | *.md)
+                echo -e "[DEBUG] ( $i ) file is a source code with a ASCII text format."
+                bad_words_sw="grep"
+                bad_words_list="./config/prohibited-words.txt"
+                if [[ ! -f $bad_words_list ]]; then
+                    echo -e "[DEBUG] Oooops. Not found $pword_list  file."
+                fi
+                bad_words_log_file="prohibited-words_result.txt"
+                bad_words_rules="--color -n -r -H -f $bad_words_list"
+
+                # Step 1: Run this module to filter prohibited words from a text file.
+                # (e.g., grep --color -f "$PROHIBITED_WORDS" $filename)
+                result_content=$($bad_words_sw $bad_words_rules $i)
+
+                # Step 2: Save a log file for debugging in case of a failure
+                echo -e "$result_content"
+                echo -e "$result_content" >  ../report/${bad_words_log_file}
+
+                # Step 3: Count prohibited words from variable result_content
+                result_count=$(echo -e "$result_content" | grep -c '^' )
+
+                # Step 4: change a value of the check result
+                if [[ $result_count -gt 0 ]]; then
+                    echo -e "Oooops. We found prohibited words: $result_count"
+                    check_result="failure"
+                else
+                    echo -e "It's okay.We did not find any prohibited words: $result_count"
+                    check_result="success"
+                fi
+                # Exit from 'for' loop statement to run just once when a commit includes text files more than 2.
+                break
```

Comment:
This module checks only one source file in commit?

Reply:
Yes. step by step. 
It means that developer has to modify correctly sequentially if 2nd source file includes prohibited words.